test/unit/puppetlabs/puppetserver/ruby/http_client_test.clj

```diff
@@ -215,25 +236,43 @@
           (is (= "200" (.runScriptlet sc "response.code")))
-          (is (= "hi" (.runScriptlet sc "response.body"))))))))
+          (is (= "hi" (.runScriptlet sc "response.body")))
+          (.runScriptlet sc "Puppet::Server::HttpClient.terminate")
+          (.terminate sc))))))
 
 (deftest clients-persist
   (testing "client persists when making HTTP requests"
     (logutils/with-test-logging
       (jetty9/with-test-webserver ring-app port
-        (let [scripting-container (create-scripting-container port)
-              client1 (.runScriptlet scripting-container "c.get('/', {}); c.client")
-              client2 (.runScriptlet scripting-container "c.post('/', 'foo', {}); c.client")]
-          (is (= client1 client2)))))))
+        (let [sc (create-scripting-container port)
+              client1 (.runScriptlet sc "c.get('/', {}); c.client")
+              client2 (.runScriptlet sc "c.post('/', 'foo', {}); c.client")]
+          (is (= client1 client2))
+          (.runScriptlet sc "Puppet::Server::HttpClient.terminate")
+          (.terminate sc))))))
 
 (deftest connections-closed
   (testing "connection header always set to close on get"
     (logutils/with-test-logging
       (jetty9/with-test-webserver ring-app-connection-closed port
-        (let [scripting-container (create-scripting-container port)]
-          (is (= "The Connection header has value close" (.runScriptlet scripting-container "c.get('/', {}).body")))))))
-  (testing "connection header always set to close on get"
+        (let [sc (create-scripting-container port)]
+          (is (= "The Connection header has value close" (.runScriptlet sc "c.get('/', {}).body")))
+          (.runScriptlet sc "Puppet::Server::HttpClient.terminate")
+          (.terminate sc)))))
+  (testing "connection header always set to close on post"
     (logutils/with-test-logging
       (jetty9/with-test-webserver ring-app-connection-closed port
-        (let [scripting-container (create-scripting-container port)]
-          (is (= "The Connection header has value close" (.runScriptlet scripting-container "c.post('/', 'foo', {}).body"))))))))
+        (let [sc (create-scripting-container port)]
+          (is (= "The Connection header has value close" (.runScriptlet sc "c.post('/', 'foo', {}).body")))
+          (.runScriptlet sc "Puppet::Server::HttpClient.terminate")
+          (.terminate sc)))))
+  (testing "master's terminate function closes the client"
+    (logutils/with-test-logging
+      (jetty9/with-test-webserver ring-app-connection-closed port
+        (let [sc (create-scripting-container port)]
+          (.runScriptlet sc "response = c.get('/', {})")
+          (is (= "200" (.runScriptlet sc "response.code")))
+          (.runScriptlet sc "require 'puppet/server/master'; Puppet::Server::Master.terminate")
+          (is (thrown? EvalFailedException
+                       (.runScriptlet sc "response = c.get('/', {})")))
```

Comment:
Does the exception you're catching here give you any more info you can use in your assertion?  e.g. does it have a nested exception of a more specific type, or if you call `getMessage` on it can you then match that string in your assertion?